### PR TITLE
Only use physical coords internally in `bevy_ui`

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

We switch back and forwards between logical and physical coordinates all over the place. Systems have to query for cameras and the UiScale when they shouldn't need to. It's confusing and fragile and new scale factor bugs get found constantly. 

## Solution

* Use physical coordinates whereever possible in `bevy_ui`. 
* Store physical coords in `ComputedNode` and tear out all the unneeded scale factor calculations and queries.
* Add an `inverse_scale_factor` field to `ComputedNode` and set nodes changed when their scale factor changes.

## Migration Guide

`ComputedNode`'s fields and methods now use physical coordinates.
`ComputedNode` has a new field `inverse_scale_factor`. Multiplying the physical coordinates by the `inverse_scale_factor` will give the logical values.

### CODE REVIEW

**Review by doup — APPROVED** [review]

**Review by BenjaminBrienen — APPROVED** [review]

File: crates/bevy_ui/src/render/mod.rs [hunk 1]
```diff
@@ -553,41 +546,26 @@ pub fn extract_default_ui_camera_view(
...
-            }),
-            Some(physical_size),
-            Some(scale_factor),
-        ) = (
-            camera.logical_viewport_size(),
-            camera.physical_viewport_rect(),
-            camera.physical_viewport_size(),
-            camera.target_scaling_factor(),
-        ) {
+        if let (Some(physical_viewport_rect),) = (camera.physical_viewport_rect(),) {
```
Comment (atlv24) [inline crates/bevy_ui/src/render/mod.rs:559]: whats the point of the tuple on either side of this let?
```suggestion
        if let Some(physical_viewport_rect) = camera.physical_viewport_rect() {
```

File: crates/bevy_ui/src/render/mod.rs [hunk 2]
```diff
@@ -762,7 +719,7 @@ pub fn extract_text_sections(
                         camera_entity: render_camera_entity.id(),
                         rect,
                         item: ExtractedUiItem::Glyphs {
-                            atlas_scaling: Vec2::splat(inverse_scale_factor),
+                            atlas_scaling: Vec2::splat(1.),
```
Comment (atlv24) [inline crates/bevy_ui/src/render/mod.rs:722]: I think there's `Vec2::ONE`

**Review by atlv24 — APPROVED** [review]
This needs a migration guide. Its a particularly sneaky type of breaking change, because it causes almost no compile errors, but underlying semantics change. @BenjaminBrienen @doup have you tested these changes/run any examples? I haven't yet.

**Review by doup — CHANGES_REQUESTED** [review]
@atlv24 I ran the UI testbed and a couple more examples. But, now I checked some other random example and there are indeed issues. 😬

So… I've run all the examples and these are the issues I've found:

- border width are different (example: `borders`, `ui_texture_atlas`)
- texture atlas slice border width (example: `ui_texture_atlas_slice`, `ui_texture_slice_flip_and_tile`, `ui_texture_slice`)
- border radius are different (example: `box_shadow`)
- text wrapping (examples: `grid` sidebar text, `scroll`, `text_debug`, `text_wrap_debug`)
- wrong text/box (?) alignment, could be because of the wrapping issue (examples: `display_and_visibility`, `text`, `size_constraints`)
- overflow clip margin (example: `overflow_clip_margin`)
- a panic! (example: `overflow_debug`)

**Comment by ickshonpe** [comment]
> @atlv24 I ran the UI testbed and a couple more examples. But, now I checked some other random example and there are indeed issues. 😬
> 
> So… I've run all the UI examples and this is what I've found:
> 
> * border width are different (example: `borders`, `ui_texture_atlas`)
> * texture atlas slice border width (example: `ui_texture_atlas_slice`, `ui_texture_slice_flip_and_tile`, `ui_texture_slice`)
> * border radius are different (example: `box_shadow`)
> * text wrapping (examples: `grid` sidebar text, `scroll`, `text_debug`, `text_wrap_debug`)
> * wrong text/box (?) alignment, could be because of the wrapping issue (examples: `display_and_visibility`, `text`, `size_constraints`)
> * overflow clip margin (example: `overflow_clip_margin`)
> * a panic! (example: `overflow_debug`)

I think all of these are fixed now.

**Review by doup — APPROVED** [review]
Everything is fine now! 🎉

**Review by atlv24 — APPROVED** [review]
ran the ui examples, all looks good, and migration guide is good!

**Review by cart — APPROVED** [review]

**Pull request merged**